src/lib.rs

```diff
@@ -482,7 +477,7 @@ fn format_lines(text: &mut StringBuffer, name: &str, config: &Config, report: &m
                         });
         }
 
-        if c == '\n' {
+        if c == '\n' || c == '\r' {
```

Comment:
I don't think this is quite right - a line ending in `\r\n` will be counted as two lines rather than one. I think it might be better to keep the original treatment, but don't increment `line_len`, although that assumes line_len is only used for the max line length test.